cmd/docs/command.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docs
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/spf13/viper"
+)
+
+const (
+	formatFlag = "format"
+)
+
+var (
+	formats = []string{"md", "man", "rst"}
+)
+
+// Command for generating flags/commands documentation.
+// It generates the documentation for all commands starting at parent.
+func Command(v *viper.Viper) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "docs",
+		Short: "Generates documentation",
+		Long:  `Generates command and flags documentation`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			for cmd.Parent() != nil {
+				cmd = cmd.Parent()
+			}
+			log.Println("Generating documentation in the current working directory")
+			switch v.GetString(formatFlag) {
+			case "md":
+				doc.GenMarkdownTree(cmd, "./")
```

Comment:
Should it also take the output dir param? You gitignored cmd/docs, which seems different.